### PR TITLE
lint: enable unused in golangci-lint

### PR DESCRIPTION
**Pull request by p0mvn**
Follow-up to: https://github.com/osmosis-labs/osmosis/issues/1897

## What is the purpose of the change

We would like to improve the quality of our codebase by trying out components of the [staticcheck](https://staticcheck.io/) linter.

This change introduced `unused` - 1st of the 4 components

Although, we added `deadcode` linter in https://github.com/osmosis-labs/osmosis/pull/1934, `unused` seems to be detecting more unused logic.

In addition to removing the unused code, the following was done:
- disable `unused` and `deadcode` for `x/gamm/pool-models/stableswap` since it is currently in development

## Documentation and Release Note

  - Does this pull request introduce a new feature or user-facing behavior changes? no
  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`? no
  - How is the feature or change documented? not applicable

### CODE REVIEW

File: x/gamm/keeper/pool_service.go [hunk 1]
```diff
@@ -183,35 +183,35 @@ func (k Keeper) JoinPoolNoSwap(
 	poolId uint64,
 	shareOutAmount sdk.Int,
 	tokenInMaxs sdk.Coins,
-) (err error) {
+) (sdk.Coins, error) {
```
Comment (p0mvn) [inline x/gamm/keeper/pool_service.go:186]: TODO: basic validation in test for the new output

File: x/gamm/keeper/msg_server.go [hunk 2]
```diff
@@ -108,22 +108,12 @@ func (server msgServer) JoinPool(goCtx context.Context, msg *types.MsgJoinPool)
...
-			types.TypeEvtPoolJoined,
-			sdk.NewAttribute(types.AttributeKeyPoolId, strconv.FormatUint(msg.PoolId, 10)),
-		),
-		sdk.NewEvent(
-			sdk.EventTypeMessage,
-			sdk.NewAttribute(sdk.AttributeKeyModule, types.AttributeValueCategory),
-			sdk.NewAttribute(sdk.AttributeKeySender, msg.Sender),
-		),
-	})
+	server.keeper.createAddLiquidityEvent(ctx, sender, msg.PoolId, addedLiquidity)
```
Comment (ValarDragon) [inline x/gamm/keeper/msg_server.go:116]: Lets please separate out lint PRs from functionality changes / more detailed refactors like this

File: x/gamm/keeper/keeper.go [hunk 3]
```diff
@@ -61,6 +61,7 @@ func NewKeeper(cdc codec.BinaryCodec, storeKey sdk.StoreKey, paramSpace paramtyp
 	}
 }
 
+// nolint: unused
 func (k *Keeper) createSwapEvent(ctx sdk.Context, sender sdk.AccAddress, poolId uint64, input sdk.Coins, output sdk.Coins) {
```
Comment (p0mvn) [inline x/gamm/keeper/keeper.go:65]: TODO: re-enable missing events

**Comment by faddat** [comment]
Strong support of this :)

**Review by mattverse — APPROVED** [review]
LGTM!

File: x/gamm/module.go [hunk 4]
```diff
@@ -87,9 +86,6 @@ type AppModule struct {
 	ak     types.AccountKeeper
 	bk     types.BankKeeper
 	keeper keeper.Keeper
-
-	accountKeeper stakingtypes.AccountKeeper
-	bankKeeper    stakingtypes.BankKeeper
```
Comment (ValarDragon) [inline x/gamm/module.go:92]: Ah good catch that this was caught

**Pull request merged**